src/vault/client/http.clj

```diff
@@ -124,10 +126,11 @@
 
 (defn- authenticate-token!
   "Updates the token ref by storing the given auth token."
-  [client token]
-  (when-not (string? token)
-    (throw (IllegalArgumentException. "Token credential must be a string")))
-  (reset! (:auth client) {:client-token (str/trim token)}))
+  [client token-map]
+  (let [token (:client-token token-map)]
+    (when-not (string? token)
+      (throw (IllegalArgumentException. "Token credential must be a string")))
+    (reset! (:auth client) {:client-token (str/trim token)})))
```

Comment:
I don't think this method should be responsible for both direct token auth and unwrapping tokens; you're losing a good bit of information here that is returned in the auth map by the unwrap call. The unwrapping flow should use the `api-auth!` method above, similar to the other login flows.